### PR TITLE
Drag & Drop dashboard

### PR DESCRIPTION
**Pull request by wmariz**
Drag & Drop dashboard

### CODE REVIEW

**Comment by wmariz** [comment]
Waiting for your feedback

**Comment by Hellowlol** [comment]
That was fast! Really Awesome work!
 Since this is just a initial commit im sure you already have noticed this but:

- If there are no active modules in the dashboard i would like if there was a welcome message like before with link to settings, forum etc.
- Add/Delete buttons seems misplaced, we need to add that some other place
- Save didnt work
- Can we change the color of the borders when it edit mode so something more pleasant for the eyes?

Thanks alot for starting on this. Its good to get some assistance from people that are more skilled in js then myself.

File: interfaces/default/js/dash.js [hunk 1]
```diff
@@ -502,22 +506,150 @@ function bytestospeed(bytes) {
...
+    $("#dash-edit").show();
+    $(".dash-row").removeClass('dash-row-edit')
+    $(".dash-module").removeClass("dash-module-edit");
+    $(".dash-row").sortable('disable');
+    $('.dash-row:empty').remove();
+    var sorted = "";
+    $(".dash-row").each(function(index) {
+        sorted += $(this).sortable("toArray") + ";"; 
+    });
+    $.get("/save_dash", 'dash_order=' + encodeURIComponent(sorted), function (data) {
```
Comment (Hellowlol) [inline interfaces/default/js/dash.js]: $.get(WEBDIR + "save_dash", 'dash_order=' + encodeURIComponent(sorted), function (data) {

**Comment by wmariz** [comment]
Hellowlol,

Can you help me with the CSS for the edit feature?

I'm not good at it.

**Comment by Hellowlol** [comment]
I'm not good at it all, but let me see what I can do. Btw, have you tested this on touch devices? I havnt done that yet.

**Comment by wmariz** [comment]
No I didn't, bit I think the drag&drag should be disabled to small devices.
Em 21/03/2015 18:26, "Hellowlol" <notifications@github.com> escreveu:

**Comment by Hellowlol** [comment]
ok I'll test it later.

**Comment by Hellowlol** [comment]
Btw I see your doing some css work that isn't needed, you should read up on boostrap 2 docs. If you want something to float right you can just add the class pull-right to the element.

**Comment by wmariz** [comment]
Thanks for pointing that.

I've tryed on my mobile phone and drag&drop dosen´t work at all.

**Comment by Hellowlol** [comment]
Ok, the Sortable library works on mobile devices. But it's no biggie I can live with this not working on mobile devices :)

**Comment by wmariz** [comment]
Or you can try with the last commit ;)

**Comment by Hellowlol** [comment]
Awesome! I don't have access toy computer now so I have to test tomorrow.

**Comment by Hellowlol** [comment]
Thanks alot! I would still like to change the colors but that have to be for some other time. GJ

**Pull request merged**